routers/web/repo/pull.go

```diff
@@ -753,11 +753,27 @@ func ViewPullFiles(ctx *context.Context) {
 	if ctx.Written() {
 		return
 	}
-	ctx.Data["CurrentReview"], err = models.GetCurrentReview(ctx.Doer, issue)
+
+	currentReview, err := models.GetCurrentReview(ctx.Doer, issue)
 	if err != nil && !models.IsErrReviewNotExist(err) {
 		ctx.ServerError("GetCurrentReview", err)
 		return
 	}
+	numPendingCodeComments := int64(0)
+	if currentReview != nil {
+		numPendingCodeComments, err = models.CountComments(&models.FindCommentsOptions{
+			Type:     models.CommentTypeCode,
+			ReviewID: currentReview.ID,
+			IssueID:  issue.ID,
+		})
+		if err != nil {
+			ctx.ServerError("CountComments", err)
+			return
+		}
+	}
+	ctx.Data["CurrentReview"] = currentReview
+	ctx.Data["PendingCodeComments"] = numPendingCodeComments
```

Comment:
```suggestion
	if currentReview != nil {
		ctx.Data["PendingCodeComments"], err = models.CountComments(&models.FindCommentsOptions{
			Type:     models.CommentTypeCode,
			ReviewID: currentReview.ID,
			IssueID:  issue.ID,
		})
		if err != nil {
			ctx.ServerError("CountComments", err)
			return
		}
	}
	ctx.Data["CurrentReview"] = currentReview
```

Reply:
This will require a extra check in the template because `.PendingCodeComments` could be nil. Also we would need `numPendingCodeComments` for pageData.

Reply:
But will the template be parsed if the server dies with an error?
If yes: Why?

Reply:
Huh? No. I mean the template currently expects `.PendingCodeComments` to be non-nil, kind of awkward to add a nill-check. https://github.com/go-gitea/gitea/pull/19612/files#diff-02d7d08e673070429b03d67df73eed87405f5b70da1b47c241d81cdeb6f0a797R4

Reply:
But where is the difference between the current approach and the proposed approach in terms of `non-nil`-ness?

Reply:
Current approach `ctx.Data["PendingCodeComments"]` will always be set(we can safely use `PendingCodeComments` in template).
Proposed approach `ctx.Data["PendingCodeComments"]` will not be set if there's no current review. (We need to check if `PendingCodeComments` is set). 

I currently prefer to always set ctx.Data, because detecting Internal server error caused by these kind of things can go unnoticed.